File: day1.js
const fs = require('fs');
const input = fs.readFileSync('day1input.txt', {encoding: 'utf8'}).split("\n");

const calcFuel = amt => {
    return Math.floor(amt / 3) - 2;
};

const fuelRequired = modules => {
    return modules.map(m => calcFuel(+m)).reduce((acc, w) => acc + w);
};

const totalFuelRequired = modules => {
    return modules.map(m => {
        let total = 0;
        let current = calcFuel(+m);
        while (current > 0) {
            total += current;
            current = calcFuel(current);
        }
        return total;
    }).reduce((acc, w) => acc + w);
};

console.log("Part 1: " + fuelRequired(input));

console.log("Part 2: " + totalFuelRequired(input));

File: day1.py
import math
from functools import reduce

input = open('day1input.txt').read().splitlines()

def calcFuel(amt):
    return math.floor(int(amt) / 3) - 2

def addFuel(f1, f2):
    return f1 + f2;

def fuelRequired(modules):
    return reduce(addFuel, map(calcFuel, modules))

def additionalFuelNeeded(fuel):
    total = 0
    current = calcFuel(int(fuel))
    while(current > 0):
        total += current
        current = calcFuel(current)
    return total

def totalFuelRequired(modules):
    return reduce(addFuel, map(additionalFuelNeeded, modules))

print("Part 1: " + str(fuelRequired(input)))
print("Part 2: " + str(totalFuelRequired(input)))
